ui/raidboss/data/06-ew/raid/p4s.ts

```diff
@@ -562,6 +958,73 @@ const triggerSet: TriggerSet<Data> = {
       condition: Conditions.caresAboutPhysical(),
       response: Responses.sharedTankBuster(),
     },
+    {
+      id: 'P4S Act Two Safe Spots',
+      type: 'Tether',
+      netRegex: NetRegexes.tether({ id: '00AD', source: 'Hesperos' }),
+      netRegexDe: NetRegexes.tether({ id: '00AD', source: 'Hesperos' }),
+      netRegexFr: NetRegexes.tether({ id: '00AD', source: 'Hespéros' }),
+      netRegexJa: NetRegexes.tether({ id: '00AD', source: 'ヘスペロス' }),
+      condition: (data) => data.act === '2',
+      // Tethers come out Cardinals (0 seconds), (3s) Other Cardinals
+      suppressSeconds: 4,
+      infoText: (data, matches, output) => {
+        const thorn = (data.thornIds ??= []).indexOf(parseInt(matches.sourceId, 16));
+        const thornMap: { [thorn: number]: string } = {
+          31: output.text!({ dir1: output.north!(), dir2: output.south!() }),
+          32: output.text!({ dir1: output.north!(), dir2: output.south!() }),
+          33: output.text!({ dir1: output.north!(), dir2: output.south!() }),
+          34: output.text!({ dir1: output.north!(), dir2: output.south!() }),
+          35: output.text!({ dir1: output.east!(), dir2: output.west!() }),
+          36: output.text!({ dir1: output.east!(), dir2: output.west!() }),
+          37: output.text!({ dir1: output.east!(), dir2: output.west!() }),
+          38: output.text!({ dir1: output.east!(), dir2: output.west!() }),
+        };
+        return thornMap[thorn];
+      },
+      outputStrings: {
+        text: {
+          en: '${dir1}/${dir2} first',
+        },
+        north: Outputs.north,
+        east: Outputs.east,
+        south: Outputs.south,
+        west: Outputs.west,
+      },
+    },
+    {
+      id: 'P4S Color Headmarkers',
+      type: 'HeadMarker',
+      netRegex: NetRegexes.headMarker({}),
+      condition: Conditions.targetIsYou(),
+      response: (data, matches, output) => {
+        // cactbot-builtin-response
+        output.responseOutputStrings = {
+          purpleTether: {
+            en: 'Purple Tether',
```

Comment:
Looks like you're right, and I also could not find a case where the tank would get marked and not the healer.

```
27|9:35:03||Healer 1|0000|E000|016F|
27|9:35:03||Tank 1|0000|E000|016F|
35|9:35:03||Healer 1||Tank 1|E000|0000|00AC|

27|9:35:03||DPS 1|0000|E000|016F|
27|9:35:03||DPS 2|0000|E000|016F|
35|9:35:03||DPS 1||DPS 2|E000|0000|00AC|

27|9:35:03||Healer 2|0000|E000|016D|
35|9:35:03||Healer 2||Tank 2|E000|0000|00AC|

27|9:35:03||DPS 3|0000|E000|016E|
27|9:35:03||DPS 4|0000|E000|016E|
35|9:35:03.9040000-05:00||DPS 3||DPS 4|E000|0000|00AC|
```

So something more would need to be done to give a callout for the purple tether-paired player there.

Some more info:
The purple headmarker goes out during Act 2, Act 4 and Curtain Call. Curtain Call is all purple, and there is the debuff that is more important for the mechanic.
Green headmarker goes out during Act 2 and Finale. The Finale callout one probably isn't necessary as it's all green tethers and I used a different trigger just to state what to do that seems to work fine (may need a durationSeconds added).
Orange headmarker goes out during Act 2.

Reply:
The other thing to keep in mind re: headmarkers is that phase 1 will have a different `firstHeadmarker` than phase 2, and so I think there will need to be some logic to handle that.  E12S has the same issue.

Reply:
The first headmarker in phase 2 is that purple headmarker.

Reply:
Thinking about it... we could work it out such that we do not need to be know the headmarker's true ID as the headmarker IDs we do see in the logs can be sorted to determine which is which and then recorded for a match later in the fight. The blue one could be predicted during Act 1.

Reply:
I think the headmarker offset should be the same across the entire zone, so you should be able to determine the true offset from a single log that has a door boss kill in it.  I know this is theoretical, but if possible I'd prefer to use true headmarker ids in the chance that eventually we can figure out a more built in way to calculate correct headmarkers.

Reply:
Ok, so looking back at my changes, I actually had the right values at one point but the problem I ran into is as you described with E12S issue. For here it's also tricky because there are three headmarkers that go out simultaneously and thus we will probably want to delay to grab all three, sort for one of them to start and go from there.